components/ocrspace/common/utils.mjs

```diff
@@ -0,0 +1,31 @@
+import fs from "fs";
+
+export const isValidUrl = (urlString) => {
+  var urlPattern = new RegExp("^(https?:\\/\\/)?" + // validate protocol
+"((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|" + // validate domain name
+"((\\d{1,3}\\.){3}\\d{1,3}))" + // validate OR ip (v4) address
+"(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*" + // validate port and path
+"(\\?[;&a-z\\d%_.~+=-]*)?" + // validate query string
+"(\\#[-a-z\\d_]*)?$", "i"); // validate fragment locator
+  return !!urlPattern.test(urlString);
+};
+
+export const checkTmp = (filename) => {
+  if (filename.indexOf("/tmp") === -1) {
+    return `/tmp/${filename}`;
+  }
+  return filename;
+};
+
+export const getUrlOrFile = (url) => {
+  if (!isValidUrl(url)) {
+    const data = fs.readFileSync(checkTmp(url));
+    const base64Image = Buffer.from(data, "binary").toString("base64");
+    return {
+      file: `data:image/jpeg;base64,${base64Image}`,
+    };
```

Comment:
I may be wrong, but I think this will only work for files with mimetype `image/jpeg`. You can get the file's mimetype if you use the `mime` package.

`import mime from "mime";`
```suggestion
    const filePath = checkTmp(url));
    const data = fs.readFileSync(filePath);
    const mimetype = mime.getType(filePath);
    const base64Image = Buffer.from(data, "binary").toString("base64");
    return {
      file: `data:${mimetype};base64,${base64Image}`,
    };
```